core/org.wso2.carbon.tomcat.ext/src/main/java/org/wso2/carbon/tomcat/ext/internal/Utils.java

```diff
@@ -35,25 +35,32 @@
     private static final String JAX_APP_PATTERN = "/jaxwebapps/";
 
 	public static String getTenantDomain(HttpServletRequest request) {
-		String requestURI = request.getRequestURI();
-		String domain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
-
-		if (!requestURI.contains("/t/")) {
-			// check for admin services - tenant admin services are deployed in
-			// super tenant flow
-			HttpSession session = request.getSession(false);
-			if (session != null && session.getAttribute(MultitenantConstants.TENANT_DOMAIN) != null) {
-				domain = (String) session.getAttribute(MultitenantConstants.TENANT_DOMAIN);
-			}
-		} else {
-			String temp = requestURI.substring(requestURI.indexOf("/t/") + 3);
-			if (temp.indexOf('/') != -1) {
-				temp = temp.substring(0, temp.indexOf('/'));
-				domain = temp;
-			}
-		}
-		return domain;
-	}
+        String requestURI = request.getRequestURI();
+        String domain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+
+        if (requestURI != null) {
+            if (!requestURI.contains("/t/")) {
+                // check for admin services - tenant admin services are deployed in
+                // super tenant flow
+                HttpSession session = request.getSession(false);
+                if (session != null && session.getAttribute(MultitenantConstants.TENANT_DOMAIN) != null) {
+                    domain = (String) session.getAttribute(MultitenantConstants.TENANT_DOMAIN);
+                }
+            } else {
+                String temp = requestURI.substring(requestURI.indexOf("/t/") + 3);
+                if (temp.indexOf('/') != -1) {
+                    temp = temp.substring(0, temp.indexOf('/'));
+                    domain = temp;
+                }
+            }
+            return domain;
+
+        } else{
+            return domain;
```

Comment:
Let's keep the default return. As it was previously. Wrapping with the if condition is enough